CMakeLists.txt

```diff
@@ -197,3 +197,5 @@ file(WRITE ${CMAKE_BINARY_DIR}/dali/__init__.py "")
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/lint.cmake)
 
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/Install.cmake)
+
+export(PACKAGE dali)
```

Comment:
I'd approach it a little bit differently: I think we should export individual targets within some namespace, e.g. using this:
```
export(TARGETS [target1 [target2 [...]]] [NAMESPACE <namespace>]
       [APPEND] FILE <filename> [EXPORT_LINK_INTERFACE_LIBRARIES])
```
There you could export `dali, dali_operators, dali_core etc...`.

Usage could be following:
```
add_subdirectory(extern/DALI)
target_link_libraries(mylib DALI::dali_core DALI::dali_kernels)
```

On top of that, we can expose convenient target that gathers every DALI target, so that user can go:
```
add_subdirectory(extern/DALI)
target_link_libraries(mylib DALI::DALI)
```

But I'd like to hear what others think about it

Reply:
you are probably right :)

Reply:
however I have no idea how to do it differently

Reply:
Changed so only the listed targets are exported